test/suite/Annotations.test.php

```diff
@@ -585,12 +590,149 @@ protected function testGetAnnotationsFromTrait()
+        $this->check($annotations[0]->note === 'inheritance-base-trait-tester', 'parent annotation first');
+        $this->check($annotations[1]->note === 'inheritance-trait', 'trait annotation second');
+
+        $annotations = Annotations::ofMethod('InheritanceTraitTester', 'traitAndChild');
+        $this->check(count($annotations) === 1, 'traitAndChild does not inherit trait');
+        $this->check($annotations[0]->note === 'inheritance-trait-tester', 'child annotation first');
+
+        $annotations = Annotations::ofMethod('InheritanceTraitTester', 'traitAndParentAndChild');
+        $this->check(count($annotations) === 2, 'traitAndParentAndChild does not inherit trait annotation');
+        $this->check($annotations[0]->note === 'inheritance-base-trait-tester', 'parent annotation first');
+        $this->check($annotations[1]->note === 'inheritance-trait-tester', 'child annotation second');
+    }
+
+    protected function testHandlesMethodAliasingWithTraits()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod('AliasTraitTester', 'baseTraitRun');
+        $this->check(count($annotations) === 2, 'baseTraitRun inherits annotation');
+        $this->check($annotations[0]->note === 'alias-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'alias-base-trait', 'non-inherited annotation goes second');
+
+        $annotations = Annotations::ofMethod('AliasTraitTester', 'traitRun');
+        $this->check(count($annotations) === 2, 'traitRun inherits annotation');
+        $this->check($annotations[0]->note === 'alias-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'alias-trait', 'non-inherited annotation goes second');
+
+        $annotations = Annotations::ofMethod('AliasTraitTester', 'run');
+        $this->check(count($annotations) === 2, 'run inherits annotation');
+        $this->check($annotations[0]->note === 'alias-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'alias-trait-tester', 'non-inherited annotation goes second');
+    }
+
+    protected function testHandlesConflictedMethodSelectionWithTraits()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod('InsteadofTraitTester', 'baseTrait');
+        $this->check(count($annotations) === 2, 'baseTrait inherits annotation');
+        $this->check($annotations[0]->note === 'insteadof-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'insteadof-base-trait-b', 'non-inherited annotation goes second');
+
+        $annotations = Annotations::ofMethod('InsteadofTraitTester', 'trate');
```

Comment:
`trait` is a reserved keyword in PHP. Do you have a better suggestion for the method name?